pages/admin/index.tsx

```diff
@@ -79,69 +71,119 @@ const COLUMNS = [
-    permitType: 'Permanent',
-    requestType: 'Replacement',
-    status: 'PENDING',
-  },
-  {
-    name: {
-      name: 'Steve Rogers',
-      id: '36565',
-    },
-    dateReceived: 'Dec 21 2021, 8:30 pm',
-    permitType: 'Permanent',
-    requestType: 'Replacement',
-    status: 'INPROGRESS',
-  },
-  {
-    name: {
-      name: 'Steve Rogers',
-      id: '36565',
-    },
-    dateReceived: 'Dec 21 2021, 8:30 pm',
-    permitType: 'Permanent',
-    requestType: 'Replacement',
-    status: 'COMPLETED',
-  },
-];
+// Application data for table
+type ApplicationData = {
+  name: {
+    firstName: string;
+    lastName: string;
+    rcdUserId: number | undefined;
+  };
+  dateReceived: Date;
+  permitType: PermitType;
+  isRenewal: boolean;
+  status: ApplicationStatus | undefined;
+};
+
+// Max number of entries in a page
+const PAGE_SIZE = 20;
 
 // Internal home page - view APP requests
 export default function Requests() {
-  const [permitTypeFilter, setPermitTypeFilter] = useControllableState({ defaultValue: 'All' });
-  const [requestTypeFilter, setRequestTypeFilter] = useControllableState({ defaultValue: 'All' });
+  //Filters
+  const [statusFilter, setStatusFilter] = useState<ApplicationStatus>();
+  const [permitTypeFilter, setPermitTypeFilter] = useState<PermitType>();
+  const [requestTypeFilter, setRequestTypeFilter] = useState<string>();
+  const [searchFilter, setSearchFilter] = useState<string>('');
+
+  // Sorting
+  const [sortOrder, setSortOrder] = useState<SortOptions>([['dateReceived', SortOrder.DESC]]);
+
+  // Debounce search filter so that it only gives us latest value if searchFilter has not been updated within last 500ms.
+  // This will avoid firing a query for each key the user presses
+  const debouncedSearchFilter = useDebounce<string>(searchFilter, 500);
+
+  // Data & pagination
+  const [requestsData, setRequestsData] = useState<ApplicationData[]>();
+  const [pageNumber, setPageNumber] = useState(0);
+  const [recordsCount, setRecordsCount] = useState(0);
+
+  // Make query to applications resolver
+  useQuery<GetApplicationsResponse, GetApplicationsRequest>(GET_APPLICATIONS_QUERY, {
+    variables: {
+      filter: {
+        order: sortOrder,
+        permitType: permitTypeFilter,
+        requestType: requestTypeFilter,
+        status: statusFilter,
+        search: debouncedSearchFilter,
+        offset: pageNumber * PAGE_SIZE,
+        limit: PAGE_SIZE,
+      },
+    },
+    notifyOnNetworkStatusChange: true,
+    onCompleted: data => {
+      setRequestsData(
+        data.applications.result.map(record => ({
+          name: {
+            firstName: record.firstName,
+            lastName: record.lastName,
+            rcdUserId: record.applicantId || undefined,
+          },
+          dateReceived: record.createdAt,
+          permitType: record.permitType,
+          isRenewal: record.isRenewal,
+          status: record.applicationProcessing?.status || undefined,
```

Comment:
```suggestion
          status: record.applicationProcessing?.status,
```
The `applicationProcessing?` should cover the `undefined` case